django-server/server/api/views.py

```diff
@@ -112,6 +112,131 @@ def get_queryset(self):
         return FixedMenuOptionReview.objects.all().filter(option_id=self.kwargs['option_pk']).order_by('id')
 
 
+class TimeStampedFilterSet(filters.FilterSet):
+    timestamp_start = filters.DateTimeFilter(required=True)
+    count = filters.NumberFilter(required=True)
+
+
+class StatsView(generics.ListAPIView):
+    serializer_class = OccupancyStatsSerializer
+    filter_backends = [filters.DjangoFilterBackend]
+    filterset_class = TimeStampedFilterSet
+
+    def get_queryset(self):
+        if getattr(self, 'swagger_fake_view', False):
+            # queryset just for schema generation metadata
+            return []
+        id = self.kwargs.get('cafeteria_pk')
+        count = self.request.query_params.get('count')
+        timestamp_start = self.request.query_params.get('timestamp_start')
+        if id is None or count is None or timestamp_start is None:
+            raise ValueError('required params not specified')
+
+        timestamp_start = parse(timestamp_start)
+        events = CameraEvent.objects.filter(cafeteria_id=id)
+        before = events.filter(timestamp__lte=timestamp_start).order_by('timestamp')
+        overrides = before.filter(event_type=CameraEvent.EventType.OCCUPANCY_OVERRIDE.value)
+        init_override_value = 0 if len(overrides) == 0 else overrides.latest('timestamp').event_value
+        changed = self.count_people(before) if len(overrides) == 0 \
+            else self.count_people(before.filter(timestamp__gt=overrides.latest('timestamp').timestamp))
```

Comment:
These lines are quite complex. Can we do something like below?
```
initial_people_count = 0;
events_before = before;

if not overrides:
    initial_people_count = ...
    events_before = ...

changed = self.count_people(events_before);
```

Reply:
rewritten in more abstract way but I think this change is included